source/speech/__init__.py

```diff
@@ -1958,6 +1958,21 @@ def getControlFieldSpeech(  # noqa: C901
 		return []
 
 
+def _getColorName(color, withRGB=False):
+	"""Helper function that will return the appropriate string presentation of a color,
+	based on whether the user wants the RGB value reported or not.
+	"""
+	if not isinstance(color, colors.RGB):
+		return str(color)
+	if withRGB:
+		return (
+			f"{color.name}"
+			f" (RGB: {color.red}, {color.green}, {color.blue}"
+			f" / #{color.red:02x}{color.green:02x}{color.blue:02x})"

```

Comment:
UX feedback: I hadn't thought about cases like `_("{color1} to {color2}")` which using this as the replacement will be very long. Perhaps it is best to have a single color format reported. Developers should be able to convert themselves. I'd personally choose Hex, since it is shorter. 

From a code design, I really don't like the idea of introducing another parameter. This should be two functions sharing a common interface. Then use something like the strategy pattern with those two implementations. Ideally, nothing in `getFormatFieldSpeech` needs to know, in reality we might need to make a change to `getFormatFieldSpeech` to enable a mechanism like this. The upside is, if we do it right we can start to split up `getFormatFieldSpeech` into a series of formatters.